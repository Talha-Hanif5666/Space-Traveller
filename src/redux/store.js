/* eslint-disable import/no-extraneous-dependencies */
import thunk from 'redux-thunk';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missions from './missions/missionsSlice';

const reducers = combineReducers({
  rockets: rocketsReducer,
  missionsReducer: missions,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
