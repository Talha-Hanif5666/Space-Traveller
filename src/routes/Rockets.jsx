import React from 'react';

function Rockets() {
  return (
    <div>
      <h1>Rockets</h1>
      <h3>Under development</h3>
    </div>
  );
}

export default Rockets;
