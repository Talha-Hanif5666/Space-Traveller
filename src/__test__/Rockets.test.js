import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Rockets from '../routes/Rockets';

describe('Mission test', () => {
  test('to render all missions: ', () => {
    const mockedRockets = [
      {
        id: 'rocket1',
        name: 'Rocket 1',
        desc: 'A rocket description',
        img: 'rocket1.jpg',
        booked: false,
      },
      {
        id: 'rocket2',
        name: 'Rocket 2',
        desc: 'A rocket description',
        img: 'rocket2.jpg',
        booked: false,
      },
    ];
    const rockets = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets rockets={mockedRockets} />
        </BrowserRouter>
      </Provider>,
    );
    expect(rockets).toMatchSnapshot();
  });
});

// import React from 'react';
// import renderer from 'react-test-renderer';
// // import {
// //   render, fireEvent, screen,
// // } from '@testing-library/react';
// import Rocket from '../components/Rocket';
// import '@testing-library/jest-dom';

// describe('calculate tests for different button inputs', () => {
//   it('renders Home element correctly', () => {
//     const rocket = {
//       id:'rocket1',
//       name: 'Mock rocket',
//       desc: 'Something about it',
//       img: 'test.jpg',
//       booked: false
//     }
//     const tree = renderer
//       .create(<Rocket rocket={rocket}/>)
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   // it('renders Quote element correctly', () => {
//   //   const tree = renderer
//   //     .create(<Quote />)
//   //     .toJSON();
//   //   expect(tree).toMatchSnapshot();
//   // });

//   // it('renders Calculator element correctly', () => {
//   //   const tree = renderer
//   //     .create(<Calculator />)
//   //     .toJSON();
//   //   expect(tree).toMatchSnapshot();
//   // });
// });
