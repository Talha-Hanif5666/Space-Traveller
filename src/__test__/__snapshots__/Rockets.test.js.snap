// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission test to render all missions:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <ul
          class="list-group list-group-flush"
        >
          <li
            class="list-group-item"
          >
            <div
              class="card mb-3"
            >
              <div
                class="row g-0"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="..."
                    class="img-fluid rounded-start"
                    src="rocket1.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Rocket 1
                    </h5>
                    <p
                      class="card-text"
                    >
                      <span
                        class="badge rounded-pill bg-primary"
                      >
                        Available
                      </span>
                      A rocket description
                    </p>
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Reserve rocket
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </li>
          <li
            class="list-group-item"
          >
            <div
              class="card mb-3"
            >
              <div
                class="row g-0"
              >
                <div
                  class="col-md-4"
                >
                  <img
                    alt="..."
                    class="img-fluid rounded-start"
                    src="rocket2.jpg"
                  />
                </div>
                <div
                  class="col-md-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Rocket 2
                    </h5>
                    <p
                      class="card-text"
                    >
                      <span
                        class="badge rounded-pill bg-primary"
                      >
                        Available
                      </span>
                      A rocket description
                    </p>
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      Reserve rocket
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
          <div
            class="card mb-3"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="..."
                  class="img-fluid rounded-start"
                  src="rocket1.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Rocket 1
                  </h5>
                  <p
                    class="card-text"
                  >
                    <span
                      class="badge rounded-pill bg-primary"
                    >
                      Available
                    </span>
                    A rocket description
                  </p>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Reserve rocket
                  </button>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          class="list-group-item"
        >
          <div
            class="card mb-3"
          >
            <div
              class="row g-0"
            >
              <div
                class="col-md-4"
              >
                <img
                  alt="..."
                  class="img-fluid rounded-start"
                  src="rocket2.jpg"
                />
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    Rocket 2
                  </h5>
                  <p
                    class="card-text"
                  >
                    <span
                      class="badge rounded-pill bg-primary"
                    >
                      Available
                    </span>
                    A rocket description
                  </p>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    Reserve rocket
                  </button>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
